invalid use of incomplete type: QPromiseConversionException in qpromise_p.h

I get a compile error with GCC 9.1.0

qpromise_p.h:586:58: error: invalid use of incomplete type 'class QtPromise::QPromiseConversionException'
             throw QtPromise::QPromiseConversionException{};
                                                                ^
qpromise_p.h:34:7: note: forward declaration of 'class QtPromise::QPromiseConversionException'
class QPromiseConversionException;


qpromise_p.h is included by qpromise_exceptions.h which contains the full definition of QPromiseConversionException, so qpromise_p.h will always be read first.

--- a/src/qtpromise/qpromise.h
+++ b/src/qtpromise/qpromise.h
@@ -8,8 +8,8 @@
 #ifndef QTPROMISE_QPROMISE_H
 #define QTPROMISE_QPROMISE_H
 
-#include "qpromise_p.h"
 #include "qpromiseexceptions.h"
+#include "qpromise_p.h"
 #include "qpromiseglobal.h"
 #include "qpromiseresolver.h"
 

--- a/src/qtpromise/qpromiseexceptions.h
+++ b/src/qtpromise/qpromiseexceptions.h
@@ -8,7 +8,6 @@
 #ifndef QTPROMISE_QPROMISEEXCEPTIONS_H
 #define QTPROMISE_QPROMISEEXCEPTIONS_H
 
-#include "qpromise_p.h"
 #include "qpromiseglobal.h"
 
 #include <QtCore/QException>
@@ -64,12 +63,6 @@ public:
         return new QPromiseUndefinedException{*this};
     }
 };
-
-// QPromiseError is provided for backward compatibility and will be
-// removed in the next major version: it wasn't intended to be used
-// directly and thus should not be part of the public API.
-// TODO Remove QPromiseError at version 1.0
-using QPromiseError = QtPromisePrivate::PromiseError;
 
 } // namespace QtPromise
 

